web/src/NetworkManager.jsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) [2022] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useEffect, useReducer } from "react";
+import { useInstallerClient } from "./context/installer";
+
+const initIpData = {
+  ip: ""
+};
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case "READ": {
+      // for now just string expected as the payload
+      // for now state struct holds just one member
+      return { ip: action.payload };
+    }
+  }
+};
+
+export default function NetworkManager() {
+  const client = useInstallerClient();
+  const [state, dispatch] = useReducer(reducer, initIpData);
+
+  useEffect(() => {
+    const first_ip = async () => {
```

Comment:
NP: in JavaScript camelCase is preferred over snake_case style. I.e., use _firstIp_ instead.